## Hypothesis: I think that students who have Computer Science as their primary major are more likely to recommend the course in the fall.


First, I will read in the csv and convert it to a column-oriented data table.

In [59]:
__author__ = "730314660"

%reload_ext autoreload
%autoreload 2

DATA_DIRECTORY="../../data"
DATA_FILE_PATH=f"{DATA_DIRECTORY}/survey.csv"

from data_utils import read_csv_rows, columnar

data_rows: list[dict[str, str]] = read_csv_rows(DATA_FILE_PATH)
data_cols: dict[str, list[str]] = columnar(data_rows)





The key is 
The key is row_number
The key is year
The key is unc_status
The key is comp_major
The key is primary_major
The key is prereqs
The key is prior_exp
The key is AP_Principles
The key is AP_A
The key is other_comp
The key is prior_time
The key is languages
The key is residency
The key is on_campus
The key is international
The key is section
The key is lesson_time
The key is sync_perf
The key is all_sync
The key is own_notes
The key is own_examples
The key is oh_visits
The key is ls_effective
The key is lsqs_effective
The key is programming_effective
The key is qz_effective
The key is oh_effective
The key is tutoring_effective
The key is kaki_effective
The key is pace
The key is difficulty
The key is understanding
The key is interested
The key is valuable
The key is would_recommend


## Now, I will select just the "comp_major" column (Yes or No) and the "would_recommend" column (scale of 1-7).

In [ ]:
from data_utils import select, head
from tabulate import tabulate

selected_data: dict[str, list[str]] = select(data_cols, ["comp_major", "would_recommend"])

tabulate(head(selected_data, 503), selected_data.keys(), "html")

## Now, I will find the total number of students majoring/minoring in Computer Science and the total number of students with other majors/minors.

In [61]:
from data_utils import columnar

i: int = 0
comp_total: int = 0
else_total: int = 0
comp_major: list[str] = selected_data["comp_major"]
would_rec: list[str] = selected_data["would_recommend"]
while i < 502:
    if comp_major[i] == "No":
        else_total += int(would_rec[i])
    else:
        comp_total += int(would_rec[i])
    i += 1
print(comp_total)
print(else_total)

1042
1803


## Next, I will count the frequency of each response.

In [62]:
from data_utils import count

comp_counts: dict[str, int] = count(selected_data["comp_major"])
print(f"comp_counts: {comp_counts}")

recommend_counts: dict[str, int] = count(selected_data["would_recommend"])
print(f"recommend_counts: {recommend_counts}")

comp_counts: {'No': 331, 'Yes - BS': 60, 'Yes - Minor': 65, 'Yes - BA': 46}
recommend_counts: {'6': 116, '7': 191, '4': 80, '3': 22, '5': 82, '1': 6, '2': 5}


## Next, I will find the average rating of recommending the course for both Comp Majors and non-Comp Majors.

In [63]:
comp_avg: float = float(comp_agg)/(502-331)
else_avg: float = float(else_agg)/331
print(comp_avg)
print(else_avg)

6.093567251461988
5.447129909365559


## Conclusion

On average, students with a Computer Science major/minor recommend taking COMP 110 more than students without a Computer Science major/minor.  On average, Comp Sci students recommend the course a 6.0936 on a 1 to 7 scale, while non Comp Sci students recommend the course a 5.4471 on a 1 to 7 scale.